### PR TITLE
Improve performance of clip_by_value when the input is IndexedSlices

### PR DESCRIPTION
**Pull request by abenmao**
Currently, the tf.clip_by_value converts the input to dense tensor and do the clipping after that. However, the gradients may be IndexedSlices when using tf.nn.embedding_lookup. It may consume large memory and make performance worse.
In this pull request, the clipping function will deal with IndexedSlices directly instead of converting to dense tensor.

### CODE REVIEW

File: tensorflow/python/ops/clip_ops_test.py [hunk 1]
```diff
@@ -96,6 +96,56 @@ def testClipIndexedSlicesByNorm(self):
     # Axes == [0, 1]
     self._testClipIndexedSlicesByNorm(values, indices, shape, 4.0, [0, 1, 2])
 
+  def _testClipTensorByValue(self, inputs, clip_value_min, clip_value_max, \
```
Comment (facaiy) [inline tensorflow/python/ops/clip_ops_test.py]: Could you move tests to [tensorflow/python/kernel_tests/clip_ops_test.py](https://github.com/tensorflow/tensorflow/blob/1a22ddc4e545ec76c0b7876788059f5a69406bca/tensorflow/python/kernel_tests/clip_ops_test.py#L54) ?

Comment (facaiy) [reply tensorflow/python/ops/clip_ops_test.py]: And also, please remove those backslashes at the end of line :-)

Comment (abenmao) [reply tensorflow/python/ops/clip_ops_test.py]: Done~

**Comment by yangjunpro** [comment]
@jhseu @vrv 
Could please take a look at this patch? It is already integrated into our internal prod environment and works fine. We think it also beneficial for the community so we made this PR.

Thanks

**Review by vrv — APPROVED** [review]
